### PR TITLE
Add special case handling for string type in `BallerinaSingletonTypeSymbol`

### LINKED ISSUE #35407: Getting incorrect value when using an external source related to `thread : "parent"`
**Description:**

**Steps to reproduce:**

Consider following code example that we used to get completions for  `thread` field as mentioned in the issue #35225 

```ballerina
public function main() {

    @strand{thread: parent}
    worker W1 {

    }
}
```

Note that here expected value is `"parent"` instead of `parent`

But if we consider following code example

```ballerina
public type Thread1 "parent" | "any";

public type StrandDataHello record {|
    string name?;
    string policy?;
    Thread1 thread = "parent";
|}; 

public const annotation StrandDataHello strand on source worker;

public function main() {

    @strand{thread: "parent"}
    worker W2 {
    }  
}

```

In this case we get the expected `"parent"`.

The reason for the issue is we are getting incorrect value when we are using an external source.

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
## Purpose

Fixes #35407

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SignatureTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.testng.annotations.Test;
+
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDefaultModulesSemanticModel;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDocumentForSingleSource;
+import static io.ballerina.tools.text.LinePosition.from;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test cases to validate signature.
+ *
```
Comment (gayalkuruppu) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SignatureTest.java:38]: Shall we add the version here?

Comment (DamithaSenevirathne) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SignatureTest.java:38]: Updated in d3d3567770a3d5d1267db3d7b4eb55ac6f74bb4f

**Review by mohanvive — APPROVED** [review]

**Review by pubudu91 — APPROVED** [review]

**Comment by pubudu91** [comment]
@DamithaSenevirathne Please tag the issues with the relevant milestone.

**Review by gayalkuruppu — APPROVED** [review]
LGTM!

**Pull request merged**